File: gitdot-api/src/resource/organization.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(ApiResource, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct OrganizationResource {
    pub id: Uuid,
    pub name: String,

    pub display_name: Option<String>,
    pub location: Option<String>,
    pub readme: Option<String>,
    pub links: Vec<String>,

    pub created_at: DateTime<Utc>,

    pub members: Option<Vec<OrganizationMemberResource>>,
}

#[derive(ApiResource, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct OrganizationMemberResource {
    pub id: Uuid,
    pub user_id: Uuid,
    pub user_name: String,

    pub role: String,
    pub role_description: Option<String>,

    pub created_at: DateTime<Utc>,
}

File: gitdot-web/packages/gitdot-api/src/resource/organization.ts
import { z } from "zod";

export const OrganizationMemberResource = z.object({
  id: z.uuid(),
  user_id: z.uuid(),
  user_name: z.string(),

  role: z.string(),
  role_description: z.string().nullable().optional(),

  created_at: z.iso.datetime(),
});
export type OrganizationMemberResource = z.infer<
  typeof OrganizationMemberResource
>;

export const OrganizationResource = z.object({
  id: z.uuid(),
  name: z.string(),

  display_name: z.string().nullable().optional(),
  location: z.string().nullable().optional(),
  readme: z.string().nullable().optional(),
  links: z.array(z.string()).default([]),

  created_at: z.iso.datetime(),

  members: z.array(OrganizationMemberResource).nullable().optional(),
});
export type OrganizationResource = z.infer<typeof OrganizationResource>;
